priv/priv.go

```diff
@@ -36,7 +36,8 @@ var (
 
 // Mutex is the snappy mutual exclusion primitive.
 type Mutex struct {
-	lock *FileLock
+	filename string
+	flock    LockedFile
```

Comment:
I think we're saying the same thing, more or less.

Let's say, just for argument's sake, that filelock is moved to its own package (I think lockfile.Lock() works perfectly for that!)

The only other bit of functionality we need from "priv" that isn't the uid==0 is WithMutex, which could be renamed WithLock, refactored to use lockfile.Lock directly, and moved to lockfile (at which point it's just a rename of priv without the uid==0 code).

And suddenly, priv.Mutex vanished. Because it's useless.